File: src/UVa/573/573.cpp

/*
*****************************************************************
*                           PROBLEMA                            *
*****************************************************************
*                       UVa 573 - The Snail                     *
*****************************************************************
*/

#include <cstdio>

int main(){

	float a, ds, dc, f, c;
	int d;

	scanf("%f %f %f %f",&a,&ds,&dc,&f);

	while(a!=0){

		d=0;
		c= 0;
		f=(f/100)*ds;
		
		do{

			d++;
			c+= ds;
			if(c>a)break;
			c-= dc;
			if(c<0) break;
			ds+=-f;
			if(ds<0) ds=0;

		}while(c>=0 && c<=a);

		if(c>=a ? printf("success on day %d\n",d):printf("failure on day %d\n",d));
		scanf("%f %f %f %f",&a,&ds,&dc,&f);
	}
	
	return 0;
}

File: src/UVa/573/573.py
#!/usr/bin/env 

'''*
*****************************************************************
*                           PROBLEMA                            *
*****************************************************************
*                       UVa 573 - The Snail                     *
*****************************************************************
*'''


def main():
	

	while True:

		h, u, d, f = input().split()
		# h, u, d, f = map(float, input().split())
		# Explicit conversion is more efficient
		h = float(h)
		u = float(u)
		d = float(d)
		f = float(f)

		if h == 0:
			break

		cur = 0
		cont = 0
		f = (f/100) * u

		while True:
			
			cont += 1
			cur += u

			if cur > h:
				print("success on day", cont)
				break

			cur -= d
			if cur < 0:
				print("failure on day", cont)
				break

			u -= f
			if u < 0:
				u = 0

if __name__ == "__main__":
	main()
